### DISCUSSION TITLE
Global middleware not triggering

### DISCUSSION THREAD

**Discussion by SamG92** [nuxt/nuxt#21789] [Questions] [answered]
Hello,

I'm trying to use a middleware with Nuxt 3 so that some pages of my site are restricted to paid subscribers.

I did some tests with global middleware and it is not working as I expected.

I define a middleware/test.global.js like that:
```
export default defineNuxtRouteMiddleware(to => {
    console.log("test")
    return
})
```
But when I navigate between pages on my site (using nuxt-links), the middleware is not triggering. It only triggers on initial page load ("reload" button on the browser for example).

Is it how it is supposed to work? How can I have code executed on every route change?

Thanks a lot
Samuel

**Comment by e112121** [comment]
maybe you need to delete return

**Reply by e112121** [reply]
if you want to  does not block navigation and will move to the next middleware function, you need to noting

**Comment by SamG92** [comment, marked as answer]
Stupid me, I was not looking at the right data and it is working as expected. Sorry!
